eng/common/scripts/Cadl-Project-Generate.ps1

```diff
@@ -0,0 +1,67 @@
+[CmdletBinding()]
+param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $ProjectDirectory
+)
+
+$ErrorActionPreference = "Stop"
+. $PSScriptRoot/Helpers/PSModule-Helpers.ps1
+. $PSScriptRoot/common.ps1
+Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+
+function NpmInstallForProject([string]$workingDirectory) {
+    Push-Location $workingDirectory
+    try {
+        $currentDur = Resolve-Path "."
+        Write-Host "Generating from $currentDur"
+        if (Test-Path "package.json") {
+            Remove-Item -Path "package.json" -Force
+        }
+        if (Test-Path ".npmrc") {
+            Remove-Item -Path ".npmrc" -Force
+        }
+        $replacementPackageJson = "$PSScriptRoot/../../emitter-package.json"
+        Write-Host("Copying package.json from $replacementPackageJson")
+        Copy-Item -Path $replacementPackageJson -Destination "package.json" -Force
+        npm install
```

Comment:
I think it might be worth adding the `--no-lock-file` option here to avoid leaving behind a lock file in the source tree as well.